adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureParser.scala

```diff
@@ -67,15 +54,28 @@ private[rdd] object GTFParser {
  */
 private[rdd] class GTFParser extends FeatureParser {
 
-  override def parse(line: String): Seq[Feature] = {
-    // skip header and comment lines
-    if (line.startsWith("#")) return Seq()
+  def isHeader(line: String): Boolean = {
+    line.startsWith("#") || line.isEmpty
+  }
+
+  override def parse(line: String,
+                     stringency: ValidationStringency): Option[Feature] = {
+
+    // skip header lines
+    if (isHeader(line)) {
+      return None
+    }
 
     val fields = line.split("\t")
-    // skip empty or invalid lines
+
+    // check for invalid lines
     if (fields.length < 8 || fields.length > 9) {
-      log.warn("Empty or invalid GTF/GFF2 line: {}", line)
-      return Seq()
+      if (stringency == ValidationStringency.STRICT) {
```

Comment:
As an aside, I feel like we've got enough of this patten (print error message if STRICT, log if LENIENT, and return None if not STRICT) around where we might as well wrap it up into a function and factor it out.